python/tank/platform/engine.py

```diff
@@ -169,6 +171,13 @@ def __init__(self, tk, context, engine_instance_name, env):
         qt.QtGui = base_def.get("qt_gui")
         qt.TankDialogBase = base_def.get("dialog_base")
 
+        qt5_root = self.__define_qt5()
+
+        if qt5_root:
```

Comment:
need some comments here please - lots of stuff happening - also I think the code is clearer if it is moved so that it's next to this bit:

```python
        # try to pull in QT classes and assign to tank.platform.qt.XYZ
        base_def = self._define_qt_base()
        qt.QtCore = base_def.get("qt_core")
        qt.QtGui = base_def.get("qt_gui")
        qt.TankDialogBase = base_def.get("dialog_base")
```

That is also happening after `init_engine` - I think unless we have good reasons to, we should keep the qt4 and qt5 *patterns* as identical as possible. Does that make sense?